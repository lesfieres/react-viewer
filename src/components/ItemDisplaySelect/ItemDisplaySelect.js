import React, { useState } from 'react'; // eslint-disable-line no-unused-vars, prop-types
import AssignmentIcon from '@material-ui/icons/Assignment';
import PropTypes from 'prop-types';
import AppsIcon from '@material-ui/icons/Apps';
import MenuIcon from '@material-ui/icons/Menu';
import MuiCheckbox from '@material-ui/core/Checkbox';
import { withStyles } from '@material-ui/core';

export const DisplayCardType = Object.freeze({
  List: 'DISPLAY_CARD_TYPE_LIST',
  SmallCard: 'DISPLAY_CARD_TYPE_SMALL_CARD',
  BigCard: 'DISPLAY_CARD_TYPE_BIG_CARD',
});

const styles = {
  buttonBox: {
    display: 'flex',
    justifyContent: 'flex-end',
  },
  checkboxRoot: {
    padding: 8,
    margin: 6,
  },
};

const prepareDisplayTypes = displayType => {
  const displayTypes = [false, false, false];
  switch (displayType) {
    case DisplayCardType.List:
      displayTypes[0] = true;
      break;
    case DisplayCardType.SmallCard:
      displayTypes[1] = true;
      break;
    case DisplayCardType.BigCard:
    default:
      displayTypes[2] = true;
      break;
  }
  return displayTypes;
};

const ItemDisplaySelect = props => {
  const { classes } = props;

  const displayTypes = [
    DisplayCardType.List,
    DisplayCardType.SmallCard,
    DisplayCardType.BigCard,
  ];
  const icons = [<MenuIcon />, <AppsIcon />, <AssignmentIcon />];
  const buttonsState = prepareDisplayTypes(props.displayType);

  return (
    <div className={classes.buttonBox}>
      {icons.map((icon, index) => (
        <MuiCheckbox
          key={index}
          classes={{
            root: classes.checkboxRoot,
            checked: classes.checkboxChecked,
          }}
          icon={icon}
          checkedIcon={icon}
          checked={buttonsState[index]}
          onChange={() => {
            props.setCurrentDisplayCardType(displayTypes[index]);
          }}
        />
      ))}
    </div>
  );
};

ItemDisplaySelect.propTypes = {
  displayType: PropTypes.string.isRequired,
  setCurrentDisplayCardType: PropTypes.func.isRequired,
};

export default withStyles(styles)(ItemDisplaySelect);
