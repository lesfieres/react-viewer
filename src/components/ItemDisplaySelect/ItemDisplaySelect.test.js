/* eslint-disable prop-types */
// eslint-disable-next-line no-unused-vars
import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import ItemDisplaySelect from './ItemDisplaySelect';

describe('ItemDisplaySelect component test', () => {
  it('renders without crashing when has the correct props', () => {
    const renderer = new ShallowRenderer();

    const props = {
      displayType: '',
      setCurrentDisplayCardType: () => {},
      classes: {},
    };

    renderer.render(<ItemDisplaySelect {...props} />);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });

  const checkCurrentButton = displayType => {
    const renderer = new ShallowRenderer();

    const props = {
      displayType,
      setCurrentDisplayCardType: () => {},
      classes: {},
    };

    renderer.render(<ItemDisplaySelect {...props} />);
    return renderer.getRenderOutput();
  };

  it('marks the List button as checked', () => {
    const result = checkCurrentButton('DISPLAY_CARD_TYPE_LIST');
    expect(result).toMatchSnapshot();
  });

  it('marks the SmallCard button as checked', () => {
    const result = checkCurrentButton('DISPLAY_CARD_TYPE_SMALL_CARD');
    expect(result).toMatchSnapshot();
  });

  it('marks the BigCard button as checked', () => {
    const result = checkCurrentButton('DISPLAY_CARD_TYPE_BIG_CARD');
    expect(result).toMatchSnapshot();
  });
});
