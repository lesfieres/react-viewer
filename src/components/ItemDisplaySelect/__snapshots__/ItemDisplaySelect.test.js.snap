// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemDisplaySelect component test marks the BigCard button as checked 1`] = `
<div>
  <WithStyles(Checkbox)
    checked={false}
    checkedIcon={<pure(MenuIcon) />}
    classes={
      Object {
        "checked": undefined,
        "root": undefined,
      }
    }
    icon={<pure(MenuIcon) />}
    onChange={[Function]}
  />
  <WithStyles(Checkbox)
    checked={false}
    checkedIcon={<pure(AppsIcon) />}
    classes={
      Object {
        "checked": undefined,
        "root": undefined,
      }
    }
    icon={<pure(AppsIcon) />}
    onChange={[Function]}
  />
  <WithStyles(Checkbox)
    checked={true}
    checkedIcon={<pure(AssignmentIcon) />}
    classes={
      Object {
        "checked": undefined,
        "root": undefined,
      }
    }
    icon={<pure(AssignmentIcon) />}
    onChange={[Function]}
  />
</div>
`;

exports[`ItemDisplaySelect component test marks the List button as checked 1`] = `
<div>
  <WithStyles(Checkbox)
    checked={true}
    checkedIcon={<pure(MenuIcon) />}
    classes={
      Object {
        "checked": undefined,
        "root": undefined,
      }
    }
    icon={<pure(MenuIcon) />}
    onChange={[Function]}
  />
  <WithStyles(Checkbox)
    checked={false}
    checkedIcon={<pure(AppsIcon) />}
    classes={
      Object {
        "checked": undefined,
        "root": undefined,
      }
    }
    icon={<pure(AppsIcon) />}
    onChange={[Function]}
  />
  <WithStyles(Checkbox)
    checked={false}
    checkedIcon={<pure(AssignmentIcon) />}
    classes={
      Object {
        "checked": undefined,
        "root": undefined,
      }
    }
    icon={<pure(AssignmentIcon) />}
    onChange={[Function]}
  />
</div>
`;

exports[`ItemDisplaySelect component test marks the SmallCard button as checked 1`] = `
<div>
  <WithStyles(Checkbox)
    checked={false}
    checkedIcon={<pure(MenuIcon) />}
    classes={
      Object {
        "checked": undefined,
        "root": undefined,
      }
    }
    icon={<pure(MenuIcon) />}
    onChange={[Function]}
  />
  <WithStyles(Checkbox)
    checked={true}
    checkedIcon={<pure(AppsIcon) />}
    classes={
      Object {
        "checked": undefined,
        "root": undefined,
      }
    }
    icon={<pure(AppsIcon) />}
    onChange={[Function]}
  />
  <WithStyles(Checkbox)
    checked={false}
    checkedIcon={<pure(AssignmentIcon) />}
    classes={
      Object {
        "checked": undefined,
        "root": undefined,
      }
    }
    icon={<pure(AssignmentIcon) />}
    onChange={[Function]}
  />
</div>
`;

exports[`ItemDisplaySelect component test renders without crashing when has the correct props 1`] = `
<div>
  <WithStyles(Checkbox)
    checked={false}
    checkedIcon={<pure(MenuIcon) />}
    classes={
      Object {
        "checked": undefined,
        "root": undefined,
      }
    }
    icon={<pure(MenuIcon) />}
    onChange={[Function]}
  />
  <WithStyles(Checkbox)
    checked={false}
    checkedIcon={<pure(AppsIcon) />}
    classes={
      Object {
        "checked": undefined,
        "root": undefined,
      }
    }
    icon={<pure(AppsIcon) />}
    onChange={[Function]}
  />
  <WithStyles(Checkbox)
    checked={true}
    checkedIcon={<pure(AssignmentIcon) />}
    classes={
      Object {
        "checked": undefined,
        "root": undefined,
      }
    }
    icon={<pure(AssignmentIcon) />}
    onChange={[Function]}
  />
</div>
`;
